kombu/transport/SQS.py

```diff
@@ -734,3 +742,41 @@ class Transport(virtual.Transport):
     @property
     def default_connection_params(self):
         return {'port': self.default_port}
+
+
+class QoS(virtual.QoS):
+    def reject(self, delivery_tag, requeue=False):
+        super().reject(delivery_tag, requeue=requeue)
+        queue_name, backoff_tasks, backoff_policy = self.extract_backoff_policy_configuration(delivery_tag)
+        if queue_name and backoff_tasks and backoff_policy:
+            self.apply_backoff_policy(queue_name, delivery_tag, backoff_policy, backoff_tasks)
+
+    def extract_backoff_policy_configuration(self,  delivery_tag):
+        queue_name = self._delivered.get(delivery_tag).delivery_info.get('routing_key')
+        if not queue_name:
+            return None, None, None
+        queue_config = self.channel.predefined_queues.get(queue_name, {})
+        backoff_tasks = queue_config.get('backoff_tasks')
+        backoff_policy = queue_config.get('backoff_policy')
+        return queue_name, backoff_tasks, backoff_policy
+
+    def apply_backoff_policy(self, queue_name, delivery_tag, backoff_policy, backoff_tasks):
+        queue_url = self.channel._queue_cache[queue_name]
+        task_name, number_of_retries = self.extract_task_name_and_number_of_retries(delivery_tag)
+        if task_name in backoff_tasks:
+            c = self.channel.sqs(queue_name)
+            c.change_message_visibility(
+                QueueUrl=queue_url,
+                ReceiptHandle=delivery_tag,
+                VisibilityTimeout=backoff_policy.get(number_of_retries)
```

Comment:
What happens if we reached the maximum? Do we just ignore the backoff?

Reply:
added handling